src/ULog.ts

```diff
@@ -223,28 +223,35 @@ export class ULog {
   }
 
   private async createIndex(): Promise<void> {
+    if (!this._header) {
+      throw new Error("Cannot createIndex before reading header");
+    }
+
     const timeIndex: IndexEntry[] = [];
     const dataCounts = new Map<number, number>();
-    let maxTimestamp = 0n;
+    let maxTimestamp: bigint | undefined;
     let logMessageCount = 0;
 
     let i = 0;
     let message: DataSectionMessage | undefined;
     while ((message = await this.readParsedMessage())) {
       if (message.type === MessageType.Data) {
-        if (message.value.timestamp > maxTimestamp) {
+        if (maxTimestamp == undefined || message.value.timestamp > maxTimestamp) {
           maxTimestamp = message.value.timestamp;
         }
         timeIndex.push([message.value.timestamp, i++, message]);
         dataCounts.set(message.msgId, (dataCounts.get(message.msgId) ?? 0) + 1);
       } else if (message.type === MessageType.Log || message.type === MessageType.LogTagged) {
-        if (message.timestamp > maxTimestamp) {
+        if (maxTimestamp == undefined || message.timestamp > maxTimestamp) {
           maxTimestamp = message.timestamp;
         }
         timeIndex.push([message.timestamp, i++, message]);
         logMessageCount++;
       } else {
-        timeIndex.push([maxTimestamp, i++, message]);
+        // Other message types do not have a timestamp field, so we use the current max. timestamp or
+        // alternatively the header timestamp which denotes when the logging started.
+        const timestamp = maxTimestamp ?? this._header.timestamp;
+        timeIndex.push([timestamp, i++, message]);
```

Comment:
Why do messages without a timestamp field need to be in the timeIndex?

Reply:
Was wondering the same. The reason is, that the current `readMessages` generator function returns all messages in the data section. If we limit this to only return message types with timestamps, then we do not have to add messages without a timestamp to the time index.

Reply:
Are these other message types that don't have timestamps used for anything? Seems like we could either not emit them from readMessages or skip them when building the index?